src/Compatibility/Core/src/iOS/EventTracker.cs

```diff
@@ -578,7 +578,7 @@ void LoadRecognizers()
 
 			if (_dragAndDropDelegate != null)
 			{
-				System.Diagnostics.Debug.Assert(OperatingSystem.IsIOSVersionAtLeast(11));
+#pragma warning disable CA1416 // TODO: 'UIDragInteraction.Delegate', 'UIView.Interactions' is only supported on: 'ios' 11.0 and later, 'maccatalyst' 11.0 and later
```

Comment:
Will it crash on iOS 10? If so, maybe we need to wrap this whole block - and related - in a OS version check.

Reply:
> Will it crash on iOS 10? 

Annotated with `SupportedOSPlatform("ios11.0")`

> maybe we need to wrap this whole block - and related - in a OS version check.

In the current code `_dragAndDropDelegate` is set only within `ios11` block, therefore the null check kind of enough if that logic will not change. 
https://github.com/dotnet/maui/blob/0575eeb77971e5420c7b14d6abab05f453f60c1a/src/Compatibility/Core/src/iOS/EventTracker.cs#L614-L629
This is actually perfect example of when we should `Assert` the platform check